libraries/ESP32/examples/Serial/RS485_Echo_Demo/RS485_Echo_Demo.ino

```diff
@@ -0,0 +1,50 @@
+/*
+  This Sketch demonstrates how to use the Hardware Serial peripheral to communicate over an RS485 bus. 
+
+  Data received on the primary serial port is relayed to the bus acting as an RS485 interface and vice versa.
+
+  UART to RS485 translation hardware (e.g., MAX485, MAX33046E, ADM483) is assumed to be configured in half-duplex
+  mode with collision detection as described in
+  https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html#circuit-a-collision-detection-circuit
+
+  To use the script open the Arduino serial monitor (or alternative serial monitor on the Arduino port). Then,
+  using an RS485 tranciver, connect another serial monitor to the RS485 port. Entering data on one terminal
+  should be displayed on the other terminal.
+*/
+
+#include <HardwareSerial.h>
+#include "hal/uart_types.h"
+
+#define RS485_RX_PIN 16
+#define RS485_TX_PIN 5
+#define RS485_RTS_PIN 37
```

Comment:
pin 37 doesn't exist on all ESP32 variants, I'd suggest moving this to GPIO 4 or other IO pin that is generally available.